### PR TITLE
Allow unstacking with mixed types

### PR DESCRIPTION
**Pull request by bramtayl**
Edit: passed

### CODE REVIEW

**Comment by bramtayl** [comment]
Ok, well, I gave up on modifying what was already there and just rewrote unstack. I've included the code below.

```julia
import Base.split

function split(grouped_frame::GroupedDataFrame, subset = :_subset)
  table = sort(grouped_frame.parent, cols = grouped_frame.cols)
  get_slice(starts, ends) = delete!( table[starts:ends, :] , grouped_frame.cols)
  summary = table[grouped_frame.starts, grouped_frame.cols]
  summary[subset] = map(get_slice, grouped_frame.starts, grouped_frame.ends)
  summary end

function join_auto(a::DataFrames.AbstractDataFrame,
                   b::DataFrames.AbstractDataFrame,
                   kind::Symbol)
  on = intersect(names(a), names(b))
  DataFrames.join(a, b, on = on, kind = kind) end

function replace_name(df::DataFrames.AbstractDataFrame, 
                      name_old::Symbol,
                      name_new::Symbol)
    fix_names = names(df)
    fix_names[fix_names .== name_old] = name_new
    names!(df, fix_names) end

function DataFrames.unstack(df::DataFrame, 
                 variable::Symbol, 
                 value::Symbol)
  
  split_df = split(groupby(df, variable))
  split_df[:_subset] =
    broadcast(replace_name, 
                split_df[:_subset], 
                  [value], split_df[variable])
  function outer_join_auto(a::DataFrame,
                           b::DataFrame)
    join_auto(a, b, :outer) end
  reduce(outer_join_auto, split_df[:_subset]) end
test = DataFrame(ID = 1:10,
                 a = 1:10,
                 b = rep("a", 10))
test_long = stack(test, [:a, :b])
test_wide = unstack(test_long, :variable, :value)
```

**Comment by bramtayl** [comment]
Ahaha!

File: src/abstractdataframe/reshape.jl [hunk 1]
```diff
@@ -211,6 +212,21 @@ function unstack(df::AbstractDataFrame, colkey::Int, value::Int)
     unstack(df2, length(df2), colkey, value)
 end
 
+import Base.convert
+
+function convert{T}(vector::AbstractVector{T})
```
Comment (quinnj) [inline src/abstractdataframe/reshape.jl]: Better to call this some other function; here you're defining `convert` on a `AbstractVector` which could lead to weirdness in people's other code that's totally unrelated.

**Comment by quinnj** [comment]
@bramtayl, any interest in reviving this? If not we can close.

**Comment by bramtayl** [comment]
Eh dunno. Given the contraints of Julia's type system, stacking and unstacking vectors of mixed types seems like a bad idea. If we're going to make it officially taboo though some documentation would be good.

**Pull request closed without merging**